### PR TITLE
[NAE-1690] Fix header responsive  overflow mode

### PR DESCRIPTION
**Pull request by machacjozef**
# Description

⚠️ extend constructor with overflowService ⚠️

Fixes [NAE-1690]

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

This was tested manually.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |      Linux Mint 20.3     |
| Runtime             |  Node v12.22.12         |
| Dependency Manager  |   NPM 6.14.16    |
| Framework version   |    Angular 13.3.1     |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @SamuelPalaj 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1690]: https://netgrif.atlassian.net/browse/NAE-1690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/header/abstract-header-service.ts [hunk 1]
```diff
@@ -81,7 +81,11 @@ export abstract class AbstractHeaderService implements OnDestroy {
     }
 
     get overflowMode(): boolean {
-        return this._overflowService.overflowMode;
+        if (this._overflowService !== null) {
```
Comment (renczesstefan) [inline projects/netgrif-components-core/src/lib/header/abstract-header-service.ts:84]: What if `_overflowService` is not null, but undefined?

Cannot be simplified to `!!this._overflowService`?

File: projects/netgrif-components-core/src/lib/panel/abstract/panel-with-immediate-data.ts [hunk 2]
```diff
@@ -15,12 +15,16 @@ import {OverflowService} from '../../header/services/overflow.service';
 export abstract class AbstractPanelWithImmediateDataComponent extends AbstractPanelWithHeaderBindingComponent implements OnDestroy {
     protected constructor(protected _translate: TranslateService,
                           protected _currencyPipe: CurrencyPipe,
-                          protected _overflowService: OverflowService) {
+                          @Optional() protected _overflowService: OverflowService) {
         super();
     }
 
     get overflowMode(): boolean {
+        if(this._overflowService !== null){
```
Comment (renczesstefan) [inline projects/netgrif-components-core/src/lib/panel/abstract/panel-with-immediate-data.ts:23]: Same as above.

Cannot be simplified to `!!this._overflowService`?

**Review by renczesstefan — DISMISSED** [review]

File: projects/netgrif-components-core/src/lib/header/abstract-header-service.ts (hunk 1 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/header/abstract-header-service.ts]: Done

File: projects/netgrif-components-core/src/lib/panel/abstract/panel-with-immediate-data.ts (hunk 2 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/panel/abstract/panel-with-immediate-data.ts]: Done

**Review by Kovy95 — APPROVED** [review]

**Pull request merged**